### PR TITLE
[ios] Refactor search bars

### PR DESCRIPTION
**Pull request by kirylkaveryn**
Closes https://github.com/organicmaps/organicmaps/issues/6758

### Issue:
There are 3 search bars in ios app now:
1. search screen
2. download maps
3. bookmarks

And there are no 2 equal search bars. 
Moreover, they are used not conventional for ios systems ways.

| Screen | Search | Bookmarks | Download Maps |
| --- | --- | --- | --- |
|  | image | image | image |
| Wat's wrong | totally custom design not matching with rest of the other searchbars. Very small elements and fonts 914 pt vs native 17pt) - hard to use (for me) | native searchbar but used in very hardcoded way with unneeded custom animations | native searchbar but implemented in place where it should not be used (as title view) and because of that totally unadjustable (looks really bad |

___

If you look at the most popular apps in ios (except google's) you will see totally native (in UI) searchbars:
GitHub
IMG_4025
Safari
IMG_4024
Telegram
IMG_4022
Photos
IMG_4023
Jira
IMG_4026

And they are using native animation which is looks like:
Simulator Screen Recording - iPhone 11 Pro - 2024-01-24 at 13 58 47

___

### Solution:
So for me, it is a good point to remove all hardcoded non-native elements and behavior to make it more consistent with the system UI/UX so users can feel comfortable when working with the app.

Custom elements that "replace native" in such components as the searchbar are really hard to debug, support, and update in the future.

Apple HIG:
https://developer.apple.com/design/human-interface-guidelines/search-fields
___
### This PR:

1. refactors all serchbars in the application with native elements and behaviour by replacing `UISearchBar` with `UISearchController` which integrated inside navigation bar properly (not hardcoded)
2. searchbar is Search screen is not replaced with UISearchController but redesigned to match its style (because it have complicated behaviour in Search screen)
3. some UI bugs fixed - see below

___

### Result:
iPhone:
| Screen | Bookmarks | Download | Search |
| --- | --- | --- | --- |
|  | image | image | image |

iPhoneSE:
image

iPad:
image

iPad on macOS:
image

Dark Theme:
image
image

___

### Animations:

There are several ways to work wit searchbar animation:
1. it can be hide but shows on drag
2. it can slides up on tap (begin searching) and navbar hides to open more free space (I disabled this for iPad - user always see navbar)
3. animation of cancel button
 
So now it looks like:

iPhone:
Simulator Screen Recording - iPhone 15 Pro - 2024-01-18 at 22 26 58
Simulator Screen Recording - iPhone 15 Pro - 2024-01-22 at 19 15 09
Simulator Screen Recording - iPhone 15 Pro - 2024-01-22 at 19 15 26

iPad:
Simulator Screen Recording - iPad Air (5th generation) - 2024-01-22 at 19 09 22
Simulator Screen Recording - iPad Air (5th generation) - 2024-01-22 at 19 11 33

___

### Additional UI Bugs are fixed during this PR
1. bug during animation on download maps page (black row at the top)
Simulator Screen Recording - iPhone 11 Pro - 2024-01-24 at 13 32 44

5.  bug with layout that causes linens at bottom of Download and Bookmark pages
before: 
image
image

after:
image
image

### CODE REVIEW

**Comment by biodranik** [comment]
Thanks! It looks better now.

1. Animations (especially in bookmarks) don't look great. Are there better options, closer to how Apple is doing animations?
2. Maybe it is worth adding a search indicator into bookmarks and downloading maps dialogs later.
3. Somehow I reproduced several times the hiding of bottom map buttons after opening and closing the menu, and after opening and closing the search dialog.

**Comment by Jean-BaptisteC** [comment]
`Closes #2593`?

**Comment by kirylkaveryn** [comment]
@biodranik 
1. Animations are 100% native.
If you are talking about nav bar height changing when pushing view controllers - this is native too. And happens every time. But apple doesn't use background color so you usually don't see it (they use transparent).

Can you please share an example of an animations you taking about?

Here examples where you can see what happens:
Notes:
image

AppStore:
image 

And in GitHub (not native but use white BG)
image

2. Search indicators are possible to implement but it needs some additional work. It's not few lines of code. New class should be created with some hacks and only for iOS > 13.0.

3. Probably this situation is not related to current PR. I don't make changes on main screen. And haven't encountered with this.
Perhaps it because button tap was misseed. When we remove tabbar we decrease the effective size of tab buttons. I can increase it in future.

File: iphone/Maps/Core/Theme/Renderers/UISearchBarRenderer.swift [hunk 1]
```diff
@@ -27,7 +27,23 @@ class UISearchBarRenderer: UIViewRenderer {
     var searchTextField: UITextField?
     if #available(iOS 13, *) {
       searchTextField = control.searchTextField
+      control.searchTextField.layer.cornerCurve = .continuous
     }
+    
+    // Default search bar implementation adds the grey transparent image for background. This code removes it and updates the corner radius. This is not working on iPad designed for mac.
```
Comment (biodranik) [inline iphone/Maps/Core/Theme/Renderers/UISearchBarRenderer.swift:33]: Good catch! Thanks for testing it on iOS and on MacOS!

**Review by biodranik — APPROVED** [review]
> `Closes #2593`?

No, these bugs are different.

> Perhaps it because button tap was misseed. When we remove tabbar we decrease the effective size of tab buttons. I can increase it in future.

This is a good hypothesis! Maybe taps are passing down to the map. But I observed that only the bottom buttons were hidden, while the zoom buttons were still visible on the map. Maybe it hints you where to look or how to reproduce the bug.

Let's merge it as is and make any improvements (if necessary) separately.

**Pull request merged**